metricshub-hardware/src/main/java/org/sentrysoftware/metricshub/hardware/threshold/FanMetricNormalizer.java

```diff
@@ -21,25 +21,135 @@
 		super(strategyTime, hostname);
 	}
 
 	/**
-	 * TODO: Complete the Javadoc for this method.
+	 * Normalizes fan speed and speed ratio limit metrics
+	 * @param monitor A given {@link Monitor}
 	 */
 	@Override
-	public void normalize(Monitor monitor) {}
+	public void normalize(Monitor monitor) {
+		normalizeSpeedLimitMetric(
+			monitor,
+			"hw.fan.speed.limit",
+			DEFAULT_LOW_DEGRADED_VALUE_SPEED_LIMIT_METRIC,
+			DEFAULT_LOW_CRITICAL_VALUE_SPEED_LIMIT_METRIC
+		);
+		normalizeSpeedLimitMetric(
+			monitor,
+			"hw.fan.speed_ratio.limit",
+			DEFAULT_LOW_DEGRADED_VALUE_SPEED_RATIO_LIMIT_METRIC,
+			DEFAULT_LOW_CRITICAL_VALUE_SPEED_RATIO_LIMIT_METRIC
+		);
+	}
+
+	/**
+	 * Normalizes the speed limit metrics.
+	 * @param monitor The monitor to normalize
+	 * @param metricNamePrefix The prefix of the metric name.
+	 * @param defaultLowDegradedValueMetric The default value for the low degraded limit metric.
+	 * @param defaultLowCriticalValueMetric The default value for the low critical limit metric.
+	 */
+	public void normalizeSpeedLimitMetric(
+		Monitor monitor,
+		final String metricNamePrefix,
+		final Double defaultLowDegradedValueMetric,
+		final Double defaultLowCriticalValueMetric
+	) {
+		if (!isMetricCollected(monitor, metricNamePrefix.replace(".limit", ""))) {
+			return;
+		}
+
+		// Get the low degraded metric
+		final Optional<NumberMetric> maybeLowDegradedMetric = findMetricByNamePrefixAndAttributes(
+			monitor,
+			metricNamePrefix,
+			Map.of("limit_type", "low.degraded")
+		);
+
+		// Get the low critical metric
+		final Optional<NumberMetric> maybeLowCriticalMetric = findMetricByNamePrefixAndAttributes(
+			monitor,
+			metricNamePrefix,
+			Map.of("limit_type", "low.critical")
+		);
+
+		// If neither low degraded nor low critical metrics are available, create both
+		if (!maybeLowDegradedMetric.isPresent() && !maybeLowCriticalMetric.isPresent()) {
+			final MetricFactory metricFactory = new MetricFactory(hostname);
+			metricFactory.collectNumberMetric(
+				monitor,
+				String.format("%s{limit_type=\"low.degraded\"}", metricNamePrefix),
+				defaultLowDegradedValueMetric,
+				strategyTime
+			);
+			metricFactory.collectNumberMetric(
+				monitor,
+				String.format("%s{limit_type=\"low.critical\"}", metricNamePrefix),
+				defaultLowCriticalValueMetric,
+				strategyTime
+			);
+		} else if (maybeLowDegradedMetric.isPresent() && maybeLowCriticalMetric.isPresent()) {
+			// If both the low degraded and low critical metrics are available, adjust the values
+			final NumberMetric lowDegradedMetric = maybeLowDegradedMetric.get();
+			final NumberMetric lowCriticalMetric = maybeLowCriticalMetric.get();
+
+			final Double lowDegradedValue = lowDegradedMetric.getValue();
+			final Double lowCriticalValue = lowCriticalMetric.getValue();
+
+			// If the low degraded value is smaller than the low critical value, swap the values
+			if (lowDegradedValue < lowCriticalValue) {
+				final Double temp = lowDegradedValue;

```

Comment:
`temp` is not required, can you please remove it from the cpu normalization as well please.